spec/sideload_spec.rb

```diff
@@ -330,6 +330,37 @@ class SideloadSpecEmployee
         expect(employees[0].positions).to eq(positions[0..1])
         expect(employees[1].positions).to eq([positions.last])
       end
+
+      context 'when match, but parent is integer and child is string' do
+        let(:positions) do
+          [
+            PORO::Position.new(id: 1, employee_id: '1'),
+            PORO::Position.new(id: 2, employee_id: '1'),
+            PORO::Position.new(id: 3, employee_id: '2')
+          ]
+        end
+
+        it 'still works' do
+          instance.assign(employees, positions)
+          expect(employees[0].positions).to eq(positions[0..1])
+          expect(employees[1].positions).to eq([positions.last])
+        end
+      end
+
+      context 'when match, but parent is string and child is integer' do
+        let(:employees) do
+          [
+            PORO::Employee.new(id: '1'),
+            PORO::Employee.new(id: '2')
+          ]
+        end
+
+        it 'still works' do
+          instance.assign(employees, positions)
+          expect(employees[0].positions).to eq(positions[0..1])
+          expect(employees[1].positions).to eq([positions.last])
+        end
+      end
```

Comment:
Would add a check for null keys not matching when they shouldn’t. Think it works here anyway but will prevent regressions